st3/sublime_lib/settings_listener.py

```diff
@@ -81,6 +127,21 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
 def on_setting_changed(
     selector: Union[str, Callable[[sublime.Settings], Selected]]
 ) -> Callable[[OnChangeListener], OnChangeListener]:
+    """
+    A decorator function to mark a method
+    of a :class:`GlobalSettingsListener` or :class:`ViewSettingsListener`
+    as a setting change handler.
+
+    The handler is not called every time the `settings` object changes,
+    but only when the value derived using the `selector` argument changes.
+    If `selector` is callable, then the derived value is ``selector(self)``.
+    If `selector` is a :class:`str`,
+    then the derived value is ``self.get(selector, None)``.
+    Otherwise, the derived value is ``projection(self, selector)``.
```

Comment:
With the "selector" concept being reused now, it might be sensible to add a section to our docs with more details and examples.